essentials/src/main/java/org/dmfs/jems/single/decorators/Preserved.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.jems.single.decorators;
+
+import org.dmfs.jems.single.Single;
+
+
+/**
+ * A {@link Single} decorator which preserves the value of the delegate and returns it instead of calling the delegate again.
+ * <p>
+ * This is meant to speed up access to potentially expensive {@link Single} implementations.
+ *
+ * @author Marten Gajda
+ */
+public final class Preserved<T> implements Single<T>
```

Comment:
I'm still not sure about the name. I mean which one of these is the better fit. I went with `Preserved` because it felt more correct.
I just came across another nice name. How about `Frozen`?

Reply:
`Frozen` could work as well, I guess. First I wrote this:

For me `Frozen` isn't that good. It suggests something more relevant change to be made by freezing. For example it is used in GoogleApi results:
https://developers.google.com/android/reference/com/google/android/gms/common/data/Freezable
and there freezing actually does something.

But then now I think it can be good, like freezing the result in place, stopping 'movement', that is delegating and processing in the delegate. So it freezes the value and the delegate in place, sort of.

Reply:
Right. "Frozen" sort of "freezes" the infinite stream of instances (which are all equal, but usually still different instances) at the very first one. Does that make sense?